Базовая стилизация всех страниц

zevbell opened the pull request:
---
:mortar_board: [Базовая стилизация всех страниц](https://up.htmlacademy.ru/htmlcss/36/user/1097219/tasks/8)

:boom: https://htmlacademy-htmlcss.github.io/1097219-sedona-36/4/

zevbell committed:
добавляю базовую стилизацию

efiand commented on images/header/navigation-button.svg:
```diff
@@ -0,0 +1,4 @@
+<svg width="160" height="36" viewBox="0 0 160 36" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Зачем нам эта кнопка? Мы стилизуем и этот текст, и этот фон.

efiand commented on index.html:
```diff
@@ -6,12 +6,14 @@
-    <header class="main-header">
-      <a class="main-link" aria-current="page">
-        <img class="main-header-logo" src="images/logo.svg" width="140" height="70" alt="Логотип сайта городка Седона.">
+    <header class="first-header">
```
Часто применяют название `page-header` для шапки всей страницы.

zevbell replied:
А, я просто как-то не подумала, что header у обеих страниц идентичный в этом случая и изгаляться не нужно.

efiand replied:
Верно.

Если он полностью разный был бы, лучше бы было изгаляться типа `page-header-index` и `page-header-inner` - для главной и внутренней, а не просто первый или второй.
Часто он частично разный (не в этом проекте) - и тогда `page-header page-header-index` и `page-header page-header-inner`

efiand commented on index.html:
```diff
@@ -43,65 +45,68 @@
-      <header class="main-hero">
-        <h1 class="visually-hidden">Седона — небольшой городок в Аризоне.</h1>
-        <img class="main-image" src="images/welcome.svg" width="458" height="352" alt="Welcome to the gorgeous Sedona, because the Grand Canyon sucks!">
+    <main class="hero">
```
Я, если честно, не очень в курсе, откуда термин `hero`, но видел его в Академии именно для блока первого "экрана", то есть в данном случае для хедера мейна.
Вообще сам по себе тут main особо и стилизации никакой не имеет, нужен ли ему класс?

efiand replied:
Будет интересно, останется ли нужным main-inner, ведь в каталоге мейн тоже без особенностей

efiand commented on index.html:
```diff
@@ -43,65 +45,68 @@
-        <img class="main-image" src="images/welcome.svg" width="458" height="352" alt="Welcome to the gorgeous Sedona, because the Grand Canyon sucks!">
+    <main class="hero">
+      <header class="hero-index">
+        <h1 class="hero-title visually-hidden">Седона — небольшой городок в Аризоне.</h1>
```
Классы выполняют стилизационную задачу и иногда задачу идентификации элемента с помощью js. Читалки и поисковые роботы никакую пользу от этого не получают.
Так как у нас нет ни той ни другой задачи, можно оставить только `visually-hidden`. Писать что-то "чтобы просто было" - плохая практика, ведущая к замусориванию проекта.

efiand commented on index.html:
```diff
@@ -43,65 +45,68 @@
-            <h3>Настоящий городок</h3>
-            <p>Седона — не аттракцион для туристов, там течёт своя жизнь</p>
-            <img src="images/photo-1.jpg" width="800" height="384" alt="Изображение гостиницы.">
+          <li class="advantages-item color-blue">
```
`color-blue` не очень удачное название: слишком общее. Если мы начинаем скатываться в атомарный CSS, то и скатываться надо целиком, то есть `background-blue color-white text-center` и т. п. классы. Но это, имхо, дичь.
Если мы называем классы не по виду, а по их назначению, как у вас до этого в проекте, то и название должно идти от назначения.
Голубой почему? Для акцентуации. `advantages-item advantages-item-accent` - например.
Почему не просто `.accent`? Потому что это слишком универсально: легко поломать проект. Да, можно договориться писать два класса рядом, типа `.advantages-item.accent {}` - но либо кто-то забудет и однажды напишет без второго класса и сломает все акценты, либо кто-то захочет переопределить поведение и не сможет, так как два класса всегда специфичнее, чем один.

efiand commented on index.html:
```diff
@@ -43,65 +45,68 @@
-            <img src="images/photo-1.jpg" width="800" height="384" alt="Изображение гостиницы.">
+          <li class="advantages-item color-blue">
+            <h3 class="advantages-title title-color">Настоящий городок</h3>
+            <p class="advantages-paragraph color-paragraph">Седона — не аттракцион для туристов, там течёт своя жизнь
```
А чем color-paragraph отличается от title-color? Только белизной текста? Зачем тогда разные классы? И почему тут color в начале названия, а там в конце, где логика?

Вообще, чтобы никогда не случилось так, чтобы в темный блок попал темный текст, а в светлый блок попал светлый текст, `всегда надо фон и цвет задавать на одном классе`.

Тогда решение может быть таким:
```css
.advantages-item-accent {
  color: #ffffff;
  background-color: #82b3d3;
}

.advantages-item-accent .advantages-paragraph {
  color: inherit;
}
```

А заголовок и так бы наследовал цвет карточки

efiand commented on index.html:
```diff
@@ -43,65 +45,68 @@
-          <li class="advantages-item">
-            <h3>Небольшая площадь</h3>
-            <p>Все достопримечательности находятся очень близко</p>
+          <li class="advantages-item color-blue-2">
```
Здесь потенцильно возможно добавление других карточек, и чтобы они повторяли цвет в шахматном порядке, можно решать проблему с помощью `nth-child`.

Тут это не воспринимается как blue, ближе к серому, да и можно запутаться в нумерациях синих. Помним о том, что какая-то карточка вообще может быть с цветом фона по умолчанию. Ведь если доп. класс потерять (при посадке сайта на CMS, например), она тут будет в цвет фона родителя, что не всегда правильно

zevbell replied:
07.02 проработаю.

efiand commented on styles/styles.css:
```diff
@@ -0,0 +1,211 @@
+  font-display: swap;
+}
+
+/* это цвет #f2f2f2 серой подложки, не уверена, что использую правильно*/
```
Да, сейчас используешь неэффективно. У нас вся страница белая, а общий фон серый. Два варианта:

1. На тег html вешаем серый фон, на body белый.
2. Оборачиваем шапку, мейн и подвал в div с классом page и вешаем белый на него. Это даже лучше, так как понадобится потом этот элемент отцентровать, ведь контейнер имеет максимальную ширину. И центровка самого body хоть и возможна, но может ломать страницу из-за браузерных расширений или работы подключаемых на страницу библиотек, например, библиотеку выбора даты jquery datepicker. Есть еще, которые ломаются, если body не на всю ширину.

И при обоих вариантах уж точно не надо потом нескольким потомкам давать белый фон, это делается в одном месте.

efiand commented on styles/styles.css:
```diff
@@ -0,0 +1,211 @@
+
+body {
+  margin: 0;
+  padding: 0;
```
Нет у него паддингов

efiand commented on styles/styles.css:
```diff
@@ -0,0 +1,211 @@
+  font-family: "PT Sans", sans-serif;
+  font-size: 22px;
+  line-height: 26px;
+  color: #000000;
```
И цвет по умолчанию и так черный

efiand commented on styles/styles.css:
```diff
@@ -0,0 +1,211 @@
+/* header */
+
+.first-header {
+  color: #000000;
```
Цвет и так черный, про фон выше написал.

efiand commented on styles/styles.css:
```diff
@@ -0,0 +1,211 @@
+  text-align: center;
+  text-decoration: none;
+  color: #000000;
+  background-color: #ffffff;
```
Не надо назначать лишний раз фон, они и так на белой подложке уже

efiand commented on styles/styles.css:
```diff
@@ -0,0 +1,211 @@
+  font-weight: 700;
+  text-align: center;
+  text-decoration: none;
+  color: #000000;
```
Более эффективно писать ссылкам `color: inherit;`. И логичнее: ведь это не сами ссылки черные, это текст тут весь черный. Сами ссылки, если и имеют цвет, то как раз отличный от основного.

efiand commented on styles/styles.css:
```diff
@@ -0,0 +1,211 @@
+.navigation-link {
+  font-size: 20px;
+  font-weight: 700;
+  text-align: center;
```
Тут бесполезное свойство.

efiand commented on styles/styles.css:
```diff
@@ -0,0 +1,211 @@
+/* main */
+
+/* не поняла про контрастный цвет фона (поставила чудовище #776e6e),
+как именно цвет должен соотноситься с изображениями? */
```
Цвет должен соотноситься так, чтобы при долгой загрузке картинки текст был виден. Эмблема в своей коричневой части почти не видна на этом фоне. С учетом того, что эмблема похитила у нас все возможные проектные цвета - голубой, белый и коричневый - думаю, как-то так можно:

![изображение](https://user-images.githubusercontent.com/26799424/216979682-3d2482a0-83c4-418e-a9c8-1c119b8b2313.png)

efiand commented on styles/styles.css:
```diff
@@ -0,0 +1,211 @@
+как именно цвет должен соотноситься с изображениями? */
+
+.hero-index {
+  color: #000000;
```
Он и так черный. Но на самом деле, так как фон темный, я бы задал тут сразу белый, вдруг заказчик захочет добавить текст. Ну и пока картинка грузится, отображается текст альта, и он бы лучше был виден белым

zevbell replied:
Я просто пока обучения смотрела, ребята методично вдалбливали, что и цвета шрифту, и фон, везде нужно добавлять. Мб я перестаралась конечно, но ок. Поняла, постараюсь учесть наследование и прочее.

efiand replied:
Мне кажется, имелось в виду, что не надо пропускать ничего по макету. Вот `#333333` в том числе и пропустила... А какими средствами это достигнуто - тут решаем мы исходя из оптимальности подходов.

efiand commented on styles/styles.css:
```diff
@@ -0,0 +1,211 @@
+  color: #000000;
+  background-color: #776e6e;
+  background-image: url(../images/main/hero-item.svg), url(../images/main/hero-background.jpg);
+  background-size: 100% auto, 100% auto;
```
С учетом возможного добавления контента в высоту я не рекомендую на растровые фоны ставить 100%, лучше `cover`

zevbell replied:
Прочитала про contain и cover. Если бы делала сама, то поставила бы по незнанию contain, потому что масштабируется без обрезания. Потому можешь, пожалуйста, относительно высоты объяснить почему cover лучше?

efiand replied:
Лучше тем, что contain может оставлять области без фона, а cover - нет. То, что немного может быть обрезано, не так важно, не контентное же изображение.
Поиграйся с тем и с другим вариантами.

zevbell replied:
Cover плашку выравнивает, contain некрасиво выделяет, а можно миксовать? Плашку 100% auto оставить, а фон cover?

efiand replied:
Миксовать можно, да. Но какая у нас задача? Макет фиксирован в 1200, и плашка и так будет нужного размера.
Если у нас задача резинового дизайна, то нам неинтересно, чтобы на большой ширине половину высоты блока эта плашка закрыла, и потому нам интереснее было бы `100% 57px`, но, чтобы оно тянулось, в саму svg надо добавить `preserveAspectRatio="none"`. В общем, там куча нюансов не с этого интенсива. А `100% auto` будет по сути мусором на этапе, когда будет готова сетка, ведь будет и без этого работать. Поэтому с заделом на будущее можно задать `100% 57px`

efiand commented on styles/styles.css:
```diff
@@ -0,0 +1,211 @@
+.hero-index {
+  color: #000000;
+  background-color: #776e6e;
+  background-image: url(../images/main/hero-item.svg), url(../images/main/hero-background.jpg);
```
Единообразие превыше всего. Любое использование `url()` по кодгайду академии предполагает кавычки, а у тебя в шрифтах они есть, а тут почему-то нет.

efiand commented on styles/styles.css:
```diff
@@ -0,0 +1,211 @@
+  background-color: #776e6e;
+  background-image: url(../images/main/hero-item.svg), url(../images/main/hero-background.jpg);
+  background-size: 100% auto, 100% auto;
+  background-repeat: no-repeat, repeat-y;
```
Повторение фона гор по вертикали не получится, оно не бесшовное. Просто `no-repeat`

efiand commented on styles/styles.css:
```diff
@@ -0,0 +1,211 @@
+  background-position: bottom center, top center;
+}
+
+.advantages h2,
```
В таких случаях стоит задумываться, а не снабжать ли одинаковые сущности одинаковыми классами? Например, `heading`?
Кроме того, мы уже как-то имели беседу, что эти тексты семантически не заголовки, а заголовки там скрытые. И если критерии академии дозволяют то, что у тебя это сейчас заголовки, то заказчик после консультации с SEO-специалистом может попросить и поменять на `<b>`. Поэтому стилизовать стоит от класса, который, кстати, у тебя там уже стоит. Это обладает меньшей специфичностью, чем вложенный тег, а чем выше специфичность, тем больше потенциальных проблем, лучше вложенностью селекторов не увлекаться.

Группировками тоже, кстати, увлекаться не стоит. Если поступила задача править все, что относится к advantages, искать по всей простыне стилей сложнее, чем если бы оно было в одном месте. Тогда ты можешь спросить, а не будет ли тогда дублей? Будут - но тут мы и приходим к понимманию, что зачем нам разные классы для одинаковых стилей? Можно назвать одинаково и поабстрактнее - `section`, `content` и т. п.

zevbell replied:
07.02 проработаю.

efiand commented on styles/styles.css:
```diff
@@ -0,0 +1,211 @@
+  font-size: 30px;
+  line-height: 36px;
+  text-transform: uppercase;
+  text-align: center;
```
А это можно повесить на секции в целом, тогда не придется у потомков отдельно задавать, будет наследование

efiand commented on styles/styles.css:
```diff
@@ -0,0 +1,211 @@
+  list-style-type: none;
+}
+
+.services-title {
```
Не может называться заголовок всей карточки и всей секции одинаково, это элементы с разными задачами. Сейчас у тебя стили работают на h2 только из-за повышения специфичности при задании стилей через вложенный тег. Но это страшный костыль. Лучше всего переименовать 4 (с учетом блока подписки) одинаковых заголовка в один, на худой конец в карточках `-title` можно переименовать на `-subtitle`

efiand commented on styles/styles.css:
```diff
@@ -0,0 +1,211 @@
+
+.services-paragraph,
+.booking-paragraph {
+  text-align: center;
```
Есть смысл наследоваться с секции, списка или как минимум с элемента списка

efiand commented on styles/styles.css:
```diff
@@ -0,0 +1,211 @@
+
+.footer-subscribe {
+  background: #776e6e url(../images/footer/subscribe-background.jpg) no-repeat center;
+  background-size: 100% auto;
```
Почему все свойства в краткой форме, а это нет?

efiand commented on styles/styles.css:
```diff
@@ -0,0 +1,211 @@
+/* footer */
+
+.footer-subscribe {
+  background: #776e6e url(../images/footer/subscribe-background.jpg) no-repeat center;
```
Страшная длинная нечитаемая строка. Рекомендую фоновые свойства писать по отдельности. Ну и багов так меньше, по опыту.

efiand commented on styles/styles.css:
```diff
@@ -0,0 +1,211 @@
+.footer-subscribe {
+  background: #776e6e url(../images/footer/subscribe-background.jpg) no-repeat center;
+  background-size: 100% auto;
+  color: #ffffff;
```
На второй странице это не перестает быть `footer-subscribe`. Но тема оформления другая. Значит, для "особого" варианта оформления этого блока можно задать доп. класс, например, `footer-subscribe-accent`

efiand commented on styles/styles.css:
```diff
@@ -0,0 +1,211 @@
+.social-media {
+  color: #000000;
+  background-color: #ffffff;
+  align-items: center;
```
Они одинаковой высоты, нет в этом необходимости

efiand commented on styles/styles.css:
```diff
@@ -0,0 +1,211 @@
+}
+
+.social-link {
+  background-size: cover;
```
Там лучше оставить их естественный размер фона, не задавая ничего. Иначе в макет можно не попасть

efiand commented on styles/styles.css:
```diff
@@ -0,0 +1,211 @@
+  color: #000000;
+  background-color: #ffffff;
+  font-size: 40px;
+  font-weight: 40px;
```
Может, line-height?

efiand commented on styles/styles.css:
```diff
@@ -0,0 +1,211 @@
+  background-color: #ffffff;
+  font-size: 40px;
+  font-weight: 40px;
+  text-align: center;
```
Бесполезное свойство. Если обеспечить размеры по величине текста извне, то зачем выравнивать. Если не делать этого - будут пустые зоны кликабельности, нехорошо

efiand commented on styles/styles.css:
```diff
@@ -0,0 +1,211 @@
+/* calalog: main */
+
+.main-inner {
+  background: #776e6e url(../images/catalog/catalog-filter-background.jpg) no-repeat center top;
```
Если вложенность страницы будет больше, хлебные крошки могут сдвинуть фильтры книзу, а они белые на белом. Тут можно использовать общий див, либо запихнуть все в хедер мейна - какой бы ни был тег, можно назвать общий враппер, например, `main-header-inner`.
И именно ему задать белый цвет, а не каждому элементу отдельно, ведь белое тут всё как раз из-за фона.

zevbell replied:
07.02 проработаю.

efiand commented on styles/styles.css:
```diff
@@ -0,0 +1,211 @@
+.main-inner {
+  background: #776e6e url(../images/catalog/catalog-filter-background.jpg) no-repeat center top;
+  background-size: 100% auto;
+  color: #000000;
```
В общем, надо прекратить везде задавать мало того что естественный браузерный цвет, так еще и на body переопределенный зачем-то.
Всё остальное так и так будет наследовать его.

zevbell replied:
07.02 проработаю.

efiand reviewed (commented):
Кроме частных замечаний, не раскрыта тема цвета `#333333`, базовой стилизации синих и коричневых кнопок и блока `number-of-likes` (хотя это все же не лайки, а число добавленных в избранное товаров)

zevbell committed:
работа над ошибками

пока без цвета #333333, базовой стилизации синих и коричневых кнопок и блока number-of-likes

zevbell committed:
добавляю кнопки

efiand commented on catalog.html:
```diff
@@ -10,260 +10,262 @@
+          <ul class="navigation-list navigation-user">
+            <li class="navigation-item">
+              <a class="navigation-link" href="#!">
+                <span class="visually-hidden">Поиск</span>
```
Нас обманули насчет точек именно в ссылках, и на следующем интенсиве, возможно, критерий поменяют, но пока что он есть. Как есть и единообразие: не все скрытые элементы, которые не имеют двоеточия или другой пунктуации, пока точкой в конце снабжены

efiand commented on catalog.html:
```diff
@@ -10,260 +10,262 @@
+              </a>
+            </li>
+            <li class="sorting-item">
+              <a class="button" href="#!">
```
По-прежнему эти две кнопки имеют не такой класс, как первая, то есть `sorting-button`

efiand commented on catalog.html:
```diff
@@ -10,260 +10,262 @@
+              <a class="pagination-link" href="#!">4</a>
+            </li>
+            <li class="pagination-item">
+              <a class="pagination-link" href="#!">...</a>
```
Здесь стилизация немного отличается, впоследствии нужно будет придумать доп. класс

efiand commented on catalog.html:
```diff
@@ -10,260 +10,262 @@
+      </main>
+      <footer class="page-footer">
+        <section class="footer-subscribe footer-subscribe-accent">
+          <h2 class="title-subscribe">Подпишитесь на рассылку</h2>
```
Это тот же `main-title`

efiand commented on catalog.html:
```diff
@@ -10,260 +10,262 @@
+      <footer class="page-footer">
+        <section class="footer-subscribe footer-subscribe-accent">
+          <h2 class="title-subscribe">Подпишитесь на рассылку</h2>
+          <p>Только полезная информация и никакого спама, честное бойскаутское!</p>
```
А это, если следовать тому же подходу, тот же `main-paragraph`, ожидаемый вместо сответствующих параграфов после `main-title`

efiand commented on index.html:
```diff
@@ -10,146 +10,149 @@
+          <h2 class="main-title">
+            Седона — небольшой городок в Аризоне, заслуживающий большего!
+          </h2>
+          <p class="advantages-paragraph">Рассмотрим причины, по которым Седона круче, чем Гранд-Каньон!</p>
```
Стили этого параграфа отличаются от стилей параграфа во внутренней карточке. То есть нужен другой класс, либо дополнительный. Но имеет смысл другой, так как в блоке сервисов такой же параграф, в блоке поиска гостиниц и в подписке - тоже. Соответственно, здесь можно поступить, как и с `main-title`, создав `main-paragraph`

efiand commented on index.html:
```diff
@@ -10,146 +10,149 @@
+              </p>
+              <img src="images/main/photo-1.jpg" width="800" height="384" alt="Изображение гостиницы.">
+            </li>
+            <li class="advantages-item advantages-item-accent-2">
```
Акцент на то и акцент, что выделяется из общей массы. В данном случае - более голубым фоном и белым текстом, а также наличием картинки.
Когда всё в акцентах - равносильно тому, что ничего не выделено.
К тому же в номерах этих запутаться легко.

Здесь карточки по базовой стилизации одинаковы, отличается только цвет фона, и можно решить его через :nth-child для четных и нечетных карточек. И в сервисах тоже.

efiand commented on index.html:
```diff
@@ -10,146 +10,149 @@
+          <h2 class="main-title">
+            Приезжайте в Седону отдохнуть в комфорте и уюте!
+          </h2>
+          <p class="services-paragraph">Опытный персонал и качественное обслуживание!</p>
```
Если есть main-title, логично тут использовать main-paragraph

efiand commented on index.html:
```diff
@@ -10,146 +10,149 @@
+          </h2>
+          <p class="services-paragraph">Опытный персонал и качественное обслуживание!</p>
+          <ul class="services-list">
+            <li class="services-item advantages-item-accent-2">
```
Если мы видим сходство с другим блоком, то это не значит, что этот блок стал "адвантажным". Это значит, что этот цвет не относится ни к "адвантажности", ни к "сервисности", а относится  к чему-то более общему.

Но тут лучше через nth-child, чтобы при добавлении новых сервисов было чередование фона в шахматном порядке. Только в advantages этот фон для четных, а тут - для нечетных.

efiand commented on index.html:
```diff
@@ -10,146 +10,149 @@
+        </section>
+        <section class="booking">
+          <h2 class="main-title">Заинтересовались?</h2>
+          <p class="booking-paragraph">Укажите предполагаемые даты поездки, и мы покажем вам лучшие предложения гостиниц
```
Если есть main-title, логично тут использовать main-paragraph

efiand commented on catalog.html:
```diff
@@ -10,260 +10,262 @@
+        </section>
+      </main>
+      <footer class="page-footer">
+        <section class="footer-subscribe footer-subscribe-accent">
```
Акцентностью пахнет все-таки на главной. Тут это стандартное черно-белое состояние

efiand commented on index.html:
```diff
@@ -10,146 +10,149 @@
+        </section>
+      </main>
+      <footer class="page-footer">
+        <section class="footer-subscribe">
```
Акцентность имеет место все-таки на главной: на внутренней стандартное черно-белое поведение.

efiand commented on styles/styles.css:
```diff
@@ -35,42 +35,29 @@ body {
-  background-color: #776e6e;
-  background-image: url(../images/main/hero-item.svg), url(../images/main/hero-background.jpg);
+  background-color: #333333;
+  background-image: url("../images/main/hero-item.svg"), url("../images/main/hero-background.jpg");
```
hero-item название неудачное, item про что-то из однородного ряда. Более понятным бы было, например, `hero-decor`

efiand commented on styles/styles.css:
```diff
@@ -35,42 +35,29 @@ body {
+  background-image: url("../images/main/hero-item.svg"), url("../images/main/hero-background.jpg");
   background-size: 100% auto, 100% auto;
-  background-repeat: no-repeat, repeat-y;
+  background-repeat: no-repeat, no-repeat;
```
Когда значения одинаковые, можно написать один раз, без дублирования: `background-repeat: no-repeat;`

efiand commented on styles/styles.css:
```diff
@@ -88,26 +75,24 @@ body {
   text-align: center;
```
Можно повесить центровку на весь advantages-item не повторять в его потомках, ведь по дизайну

efiand commented on styles/styles.css:
```diff
@@ -88,26 +75,24 @@ body {
 }
 
-.color-blue-2 {
+.advantages-item-accent-2 {
```
Это лучше сделать через `.advantages-item:nth-child(even)`

efiand commented on styles/styles.css:
```diff
@@ -88,26 +75,24 @@ body {
 }
 
-.color-blue-3 {
+.advantages-item-accent-3 {
```
Это лучше сделать через `.advantages-item:nth-child(odd)`

efiand commented on styles/styles.css:
```diff
@@ -122,53 +107,56 @@ body {
 
-.services-paragraph,
-.booking-paragraph {
+.services p,
```
Снова группировка, которая размазывает стили booking по разным местам простыни, усложняя поддержку, если задача будет на доработку именно booking.
Идея с main-paragraph не так уж и плоха была бы тут.

efiand commented on styles/styles.css:
```diff
@@ -122,53 +107,56 @@ body {
 
 .footer-subscribe {
-  background: #776e6e url(../images/footer/subscribe-background.jpg) no-repeat center;
+  background-image: url("../images/footer/subscribe-background.jpg");
```
Все-таки это к footer-subscribe-accent, который вешаем на главную.
Иначе придется писать двойные стили: под акцент перебивая это свойство. Хотя можно было просто не писать для того, для чего оно не надо

efiand commented on styles/styles.css:
```diff
@@ -122,53 +107,56 @@ body {
+  background-image: url("../images/footer/subscribe-background.jpg");
   background-size: 100% auto;
+  background-repeat: no-repeat;
+  background-position: center;
```
https://up.htmlacademy.ru/htmlcss/36/criteries#TEST-04

Для растровых **небесшовных** фонов советую забыть все кроме cover, ибо иначе возможна нехватка фона при переполнении:
![изображение](https://user-images.githubusercontent.com/26799424/217500605-16ead18a-c7dd-401e-a332-82c084bc5861.png)

А, используя `background-size: cover`, мы не нужнаемся более в `background-repeat: no-repeat;` и 
`background-position: center;`

efiand commented on styles/styles.css:
```diff
@@ -122,53 +107,56 @@ body {
   color: #ffffff;
 }
 
+.footer-subscribe p {
```
Такой же параграф, как под main-title в других местах

efiand commented on styles/styles.css:
```diff
@@ -181,9 +169,10 @@ body {
 .main-inner {
-  background: #776e6e url(../images/catalog/catalog-filter-background.jpg) no-repeat center top;
+  background-image: url("../images/catalog/catalog-filter-background.jpg");
   background-size: 100% auto;
```
cover, а background-position и background-repeat теряют смысл. Но самое главное - фон надо вешать не на main, и вот почему:

![изображение](https://user-images.githubusercontent.com/26799424/217502242-bfaf8ded-0671-420f-8bed-cfc317f11573.png)

заказчик попросил добавить фильтров (на защите эта ситуация вот - https://up.htmlacademy.ru/htmlcss/36/criteries#TEST-04), и белое оказалось на белом. Поэтому нужно обернуть хедер мейна и фильтры в общий див, или положить форму фильтров в header мейна. И вот ему задать белый текст и фоны, а на потомках белый текст задавать уже не нужно

efiand commented on styles/styles.css:
```diff
@@ -209,3 +198,7 @@ body {
 }
+
+.footer-subscribe-accent {
+  background-image: none;
```
А можно было просто не задавать по умолчанию. То есть задать именно на акцентный блок. Подумай о значении слова "акцентный" в контексте моих рекомендаций

efiand commented on catalog.html:
```diff
@@ -114,8 +114,8 @@ <h2 class="visually-hidden">Выбор гостиницы.</h2>
             </fieldset>
-            <button type="submit">Применить</button>
-            <button type="reset">Сбросить</button>
+            <button class="button-submit" type="submit">Применить</button>
```
Так как общее во всех кнопках есть (шрифт, отступы, поведение при наведениях) - я рекомендую и общий класс `.button`

efiand commented on index.html:
```diff
@@ -54,71 +54,72 @@ <h1 class="visually-hidden">Седона — небольшой городок 
             <li class="advantages-item advantages-item-accent-2">
               <h3 class="advantages-title">Небольшая площадь</h3>
-              <p class="advantages-paragraph">Все достопримечательности находятся очень близко</p>
+              <p class="advantages-paragraph">Все достопримечательности<br> находятся очень близко</p>
```
Лучше br добавлять после создания сеток - там, где ширина ячеек не сделала дело автоматически. Так будет меньше работы

efiand commented on styles/styles.css:
```diff
@@ -15,7 +15,6 @@
 }
 
 body {
-  margin: 0;
```
У body нужно сбросить маргины

efiand commented on styles/styles.css:
```diff
@@ -46,15 +43,64 @@ body {
   color: inherit;
 }
 
+.navigation-search::before {
```
Обрати вниммание, что ты повторяешь кучу свойств в двух правилах. Это наводит на мысль, что есть некий общий класс для before - например, navigation-user-link. И тут настало время, когда я смогу объяснить, почему navigation-link не нужен на абы какие ссылки, а нужен на ссылки только с определенным стилем. CSS и классы это же про стили. То есть тут другой общий класс нужен, не navigation-link

efiand commented on styles/styles.css:
```diff
@@ -46,15 +43,64 @@ body {
+}
+
+.number-of-selections {
+  display: flex;
```
Тут на нужно столько усложнять. Банальные паддинги и всё

efiand commented on styles/styles.css:
```diff
@@ -46,15 +43,64 @@ body {
+}
+
+.navigation-button {
+  display: flex;
```
Банальные паддинги и text-align у кнопки и всё

efiand commented on styles/styles.css:
```diff
@@ -46,15 +43,64 @@ body {
+
+.navigation-button {
+  display: flex;
+  flex-direction: row;
```
Это и так дефолтное значение, не надо везде писать

efiand commented on styles/styles.css:
```diff
@@ -46,15 +43,64 @@ body {
+  background-color: #756157;
+  border-radius: 4px;
+  width: 160px;
+  height: 36px;
```
Не задавай жесткую высоту элементам с текстом, плохо это. Текст вылезет, если его добавят. Делай отступы паддингами в таких случаях

efiand commented on styles/styles.css:
```diff
@@ -46,15 +43,64 @@ body {
+  height: 36px;
+  text-transform: uppercase;
+  padding: 8px 50px;
+  margin: 14px 70px 14px 20px;
```
Маргины не надо задавать кнопке, надо паддинги самому родителю.
Вот пример для общего понимания:
![изображение](https://user-images.githubusercontent.com/26799424/217504640-c3177472-6acd-40fc-9e41-3868f4de25e8.png)

efiand commented on styles/styles.css:
```diff
@@ -112,23 +158,70 @@ body {
 
+.booking-button {
+  display: flex;
+  flex-direction: row;
```
Где ты набралась этого антипаттерна делать флекс на кнопки? К тому же смотри сколько свойств ты повторяешь, хотя можно всем текстовым кнопкам задать общий класс `button`

efiand commented on styles/styles.css:
```diff
@@ -112,23 +158,70 @@ body {
+}
+
+.subscribe-button {
+  display: flex;
```
И тут флексы не нужны, отступы делаем только паддингами

efiand commented on styles/styles.css:
```diff
@@ -112,23 +158,70 @@ body {
-  color: #000000;
-  background-color: #ffffff;
+.field {
+  display: flex;
```
И тут не нужны флексы

efiand commented on styles/styles.css:
```diff
@@ -140,16 +233,16 @@ body {
-.social-link-vk {
-  background-image: url("../images/footer/social-vk.svg");
+.social-link-vk::before {
+  content: url("../images/footer/social-vk.svg");
```
Контент озвучивается читалкой, нужно фон либо маску

efiand reviewed (commented):
Общий момент
![изображение](https://user-images.githubusercontent.com/26799424/217505530-d7e99f73-6ad3-4409-969c-b90105bce82f.png)

- Удалить мусор
- Рекомендую перенести свойства со стрелками в общий класс `.button`, чтобы везде не дублировать их
